src/hooks/useHandleModalClick.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from "react";
+
+export default function useHandleModalClick(modalRef, handleModalClose) {
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      console.log(modalRef.current);
+      if (modalRef.current && !modalRef.current.contains(event.target)) {
+        console.log("asdf");
+        //이 부분 잘 한걸까..
+        if (
+          modalRef.current.id === "popOver" &&
+          event.target.id === "moreBtn"
+        ) {
+          console.log("asdfasdjfla;ksdjf;l");
+          return;
+        }
+        handleModalClose();
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [modalRef, handleModalClose]);
```

Comment:
P3:
useEffect안에 바로 함수를 작성하셔도 좋지만 아래와 같이 분리하시면 가독성이 더 좋을 것 같습니다.

```suggestion
   const handleClickOutside = useCallback(
    (event) => {
      if (modalRef.current && !modalRef.current.contains(event.target)) {
        //이 부분 잘 한걸까..
        if (
          modalRef.current.id === "popOver" &&
          event.target.id === "moreBtn"
        ) {
          return;
        }
        handleModalClose();
      }
    },
    [handleModalClose, modalRef]
  );

  useEffect(() => {
    document.addEventListener("mousedown", handleClickOutside);

    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, [handleClickOutside]);
```